microprofile/weld/weld-core-impl/src/main/java/org/jboss/weld/bean/proxy/ProxyFactory.java

```diff
@@ -632,7 +632,10 @@ private Class<T> createProxyClass(Class<?> originalClass, String proxyClassName)
 
     private ClassFile newClassFile(String name, int accessFlags, String superclass, String... interfaces) {
         try {
-            return new ClassFile(name, accessFlags, superclass, interfaces);
+            // We need to use a (non-deprecated) method that avoids instantiating DefaultClassFactory.INSTANCE
+            // If that happens, we will have module accessibility issues and the need to use --add-opens clausules
```

Comment:
Typo: `clausules`

Reply:
This is an exact copy from the source code of Weld, I want to keep our changes located in a single section in the file. So I will not fix this.

Reply:
see https://github.com/weld/core/blob/d6729a0bd53b85d7dc3177405a6da8724ec80bb5/impl/src/main/java/org/jboss/weld/bean/proxy/ProxyFactory.java#L517